ocaml/lambda/translcore.ml

```diff
@@ -127,14 +127,22 @@ let transl_apply_position position =
     if Config.stack_allocation then Rc_close_at_apply
     else Rc_normal
 
+let iter_exclaves ~f lam =
+  let rec loop ~depth = function
+    | Lregion (body, _layout) ->
+      loop ~depth:(depth + 1) body
+    | Lexclave excl when depth = 0 -> f excl
+    | Lexclave excl ->
+      loop ~depth:(depth - 1) excl
+    | lam ->
+      shallow_iter lam
+        ~tail:(loop ~depth)
+        ~non_tail:ignore (* Exclaves are in tail position *)
+  in
+  loop ~depth:0 lam
```

Comment:
I think this part is no different than the original code other than supporting nested `Lregion`. Note that nested regions are not currently supported and would break in a few places in lambda, so this is isn't strictly necessary.